python/ray/dag/class_node.py

```diff
@@ -85,23 +83,25 @@ def __getattr__(self, method_name: str):
             raise AttributeError(f".bind() cannot be used again on {type(self)} ")
         # Raise an error if the method is invalid.
         getattr(self._body, method_name)
-        call_node = _UnboundClassMethodNode(self, method_name)
+        call_node = _UnboundClassMethodNode(self, method_name, {})
         return call_node
 
     def __str__(self) -> str:
         return get_dag_node_str(self, str(self._body))
 
 
 class _UnboundClassMethodNode(object):
-    def __init__(self, actor: ClassNode, method_name: str):
+    def __init__(self, actor: ClassNode, method_name: str, options: dict):
+        # TODO(sang): Theoretically, We should use weakref cuz it is
+        # a circular dependency but when I used weakref, it fails
+        # because we cannot serialize the weakref.
```

Comment:
Explain the concern about circular dependencies here?

Reply:
you can add a simple wrapper for `weakref` to address the serialization issue if it's needed. I did the same in a patch to the `pydantic` repo:
https://github.com/pydantic/pydantic/pull/7780